core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -83,10 +83,10 @@ message TaskDataWrapper {
   int64 launch_time = 5;
   int64 result_fetch_start = 6;
   int64 duration = 7;
-  string executor_id = 8;
-  string host = 9;
-  string status = 10;
-  string task_locality = 11;
+  optional string executor_id = 8;
```

Comment:
Have we considered defining a message as follows? 

```
message NullableString {
  bool is_null = 1;
  string value = 2;
}
```
In this way, we can distinguish the string that are originally optional, but we need to record one more state

Reply:
Yeah that's complicated